.circleci/config.yml

```diff
@@ -72,6 +100,64 @@ jobs:
       - store_test_results:
           path: /go/out/tests
 
+  e2e-simple-cni:
+    <<: *integrationDefaults
+    environment:
+            - KUBECONFIG: /go/out/minikube.conf
+            - TEST_ENV: minikube-none
+            - GOPATH: /go
+    steps:
+      - <<: *initWorkingDir
+      - checkout
+      - attach_workspace:
+          at:  /go
+      - <<: *markJobStartsOnGCS
+      - run: make sync
+      - run: make build
+      - run:
+          command: |
+            if [ ! -f /go/out/linux_amd64/release/istio-cni ]; then
+              # Should only happen when re-running a job, and the workspace is gone
+              time make build
+            fi
+            make docker.all
+      - run: docker images
+      - run:
+          name: Get Istio
+          shell: /bin/bash
+          working_directory: /go/src/istio.io/istio
+          command: |
+            git clone https://github.com/tiswanso/istio.git .
```

Comment:
Since the Circle CI is not going to pass anyway you might want to change this istio/istio and checkout master below

Reply:
Just a thought. Instead of duplicating a lot of stuff for e2e or integration tests, maybe these tests can be incorporated into the istio repo. This repo will simply perform unit tests. 

In that case, istio.io/cni will be brought into istio.io/istio as a vendor dependency.

Reply:
@baodongli - we aren't duplicating anything we are just using istio/istio e2e directly with a few tweaks.   Most of the tweaks would be required even without e2e since they are needed to have a CNI deploy option.   I think the approach above is cleaner than having a vendor dependency because it constrains the dependency to only the gate testing.   The difference might be subtle though so either approach could be workable.

Reply:
This comment thread really should be on #33 ... this PR is just rebased on top of it

Reply:
Yeah, I realized that. A few PRs contain similar change.

Reply:
Wonder if you can just modify your PR so that only changes pertaining to this PR is included. Barring changes in the same file from other PRs, When you commit, you can just commit your change only, and push them up. It seems to be possible in your case.

Reply:
Yeah, I usually do that but have it on top of this to get e2e_simple to pass.  But circleci isn't hooked up properly now after the repo move and, as @john-a-joyce points out, the e2e test isn't using the right tag anyway for each specific istio/cni change.  I'll go ahead and rebase it on master to get rid of the other diffs.